src/app/cards/add/Form.tsx

```diff
@@ -0,0 +1,293 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import { ChangeEvent, useState } from 'react';
+
+import { TopNav } from '@/components/TopNav';
+import { convertImgToBase64 } from '@/lib/imageConvertor';
+import { IType } from '@/models/Type';
+import CardAdaptor from '@/services/CardAdaptor';
+
+export function Form({
+    handleCreateCard,
+    getTypesResult,
+}: {
+    handleCreateCard: ({
+        image_front,
+        image_back,
+        name,
+        type_id,
+        number,
+        note,
+    }: {
+        image_front: string;
+        image_back: string;
+        name: string;
+        type_id: string;
+        number: string;
+        note: string;
+    }) => Promise<{ id?: string; error?: string }>;
+    getTypesResult: {
+        types?: IType[];
+        error?: any;
+    };
+}) {
+    const router = useRouter();
+
+    const [card, setCard] = useState<{
+        image_front: string;
+        type_id: string;
+        image_back?: string;
+        name?: string;
+        number?: string;
+        note?: string;
+    }>({ image_front: '', type_id: '' });
+    const [progress, setProgress] = useState(0);
+    const [loading, setLoading] = useState(false);
+    const [errorMsg, setErrorMsg] = useState('');
+
+    function handleImgFrontChange(event: ChangeEvent<HTMLInputElement>) {
+        const file = event.target.files && event.target.files[0];
+        if (!file) {
+            setCard({ ...card, image_front: '' });
+            return;
+        }
+        const reader = new FileReader();
+        convertImgToBase64(
+            reader,
+            file,
+            ({ resizedImgStr, error }: { resizedImgStr?: string; error?: string }) => {
+                if (!resizedImgStr) {
+                    console.log(error);
+                    setCard({ ...card, image_front: '' });
+                    return;
+                }
+                setCard({ ...card, image_front: resizedImgStr });
+            }
+        );
+    }
+
+    function handleImgBackChange(event: ChangeEvent<HTMLInputElement>) {
```

Comment:
[NIT] handleImgBackChange and handleImgFrontChange are very similar. It will be better if you can use one function to replace them.